app/bundles/CampaignBundle/Form/Type/EventType.php

```diff
@@ -43,6 +57,30 @@ public function buildForm(FormBuilderInterface $builder, array $options)
             ]
         );
 
+        if (in_array($options['data']['eventType'], ['action']) && in_array($options['data']['type'], ['lead.changepoints'])) {
```

Comment:
This should not be hard coded here rather be part of Mautic\LeadBundle\Form\Type\PointActionType and the other forms applicable. If it's used in more than one place, use a trait or something. 

I realize this means that you cannot have an association on the campaign events table (so the migration and association should be removed from CampaignEvent entity). But this is by design. Features implemented in campaigns via other bundles should not hard code their support into the campaign code. Rather leverage the existing event listeners. 

Instead of doing an association with campaign events, you'll need to simply check that the category still exists when processing the action in the appropriate action processing listener.

Reply:
Sorry we didn't understand what you're expecting from us on this topic. We wanted to let users being able to select right from the campaign builder, in the specifi step which category was affected (like we can say send an email > this template)

so we needed to extend the update score actions to let the user choose the score category right ?